code/ATMOSPHERICS/components/binary_devices/circulator.dm

```diff
@@ -71,6 +74,43 @@
 	..()
 	update_icon()
 
+/obj/machinery/atmospherics/binary/circulator/proc/get_inlet_air()
+	if(sides_inverted==0)

```

Comment:
`if(side_inverted == 0)`

Look out for this jamming together throughout the rest of your PR.